packages/nuxt/src/core/runtime/nitro/renderer.ts

```diff
@@ -247,6 +247,9 @@ export default defineRenderHandler(async (event): Promise<Partial<RenderResponse
   }
 
   const _rendered = await renderer.renderToString(ssrContext).catch(async (error) => {
+    // We use error to bypass full render if we have an early response we can make
+    if (ssrContext._renderResponse && error.message === 'skipping render') { return {} as ReturnType<typeof renderer['renderToString']> }
```

Comment:
(please hold on to this. returning empty is probably unsafe while checking https://github.com/nuxt/nuxt/issues/21414)

Reply:
We do not return an empty response - we return the value of `ssrContext._renderResponse` below on L260.

Reply:
do you still want to defer this PR?

Reply:
Seems good if we are not returning an empty object to the nitro renderer :)